metadata_service/proxy/atlas_proxy.py

```diff
@@ -93,43 +94,78 @@ def _get_table_entity(self, *, table_uri: str) -> Tuple[EntityUniqueAttribute, D
                 table_info['entity'],
                 qualifiedName=table_info.get('name')), table_info
         except Exception as ex:
-            LOGGER.exception('Table not found. {}'.format(str(ex)))
+            LOGGER.exception(f'Table not found. {str(ex)}')
             raise NotFoundException('Table URI( {table_uri} ) does not exist'
                                     .format(table_uri=table_uri))
 
+    def _get_column(self, *, table_uri: str, column_name: str) -> Dict:
+        """
+        Fetch the column information from referredEntities of the table entity
+        :param table_uri:
+        :param column_name:
+        :return: A dictionary containing the column details
+        """
+        try:
+            table_entity, _ = self._get_table_entity(table_uri=table_uri)
+            columns = table_entity.entity['attributes'].get('columns', list())
+            for column in columns:
+                col_details = table_entity.referredEntities[column['guid']]
+                if column_name == col_details['attributes']['qualifiedName']:
+                    return col_details
+
+            raise NotFoundException(f'Column not found: {column_name}')
+
+        except KeyError as ex:
+            LOGGER.exception(f'Column not found: {str(ex)}')
+            raise NotFoundException(f'Column not found: {column_name}')
+
     def get_user_detail(self, *, user_id: str) -> Union[UserEntity, None]:
         pass
 
     def get_table(self, *, table_uri: str) -> Table:
         """
         Gathers all the information needed for the Table Detail Page.
         :param table_uri:
-        :return:
+        :return: A Table object with all the information available
+        or gathered from different entities.
         """
         entity, table_info = self._get_table_entity(table_uri=table_uri)
         table_details = entity.entity
 
         try:
             attrs = table_details['attributes']
-            rel_attrs = table_details['relationshipAttributes']
 
             tags = []
-            for classification in table_details.get("classifications", list()):
+            # Using or in case, if the key 'classifications' is there with a None
+            for classification in table_details.get("classifications") or list():
                 tags.append(
                     Tag(
                         tag_name=classification.get('typeName'),
                         tag_type="default"
                     )
                 )
 
+            columns = []
+            for column in attrs.get('columns') or list():
+                col_entity = entity.referredEntities[column['guid']]
+                col_attrs = col_entity['attributes']
```

Comment:
does attributes key always exist in col_entity?

Reply:
True. Each entity in Apache Atlas has the `attributes` key for sure.